frontend/src/utilities/useEnableApplication.tsx

```diff
@@ -68,17 +63,27 @@ export const useEnableApplication = (
         if (isInternalRouteIntegrationsApp(internalRoute)) {
           getIntegrationAppEnablementStatus(internalRoute)
             .then((response) => {
-              if (!response.isInstalled && response.canInstall) {
+              if (
+                response.isInstalled &&
+                response.canInstall &&
+                response.variablesValidationStatus === ''
+              ) {
                 watchHandle = setTimeout(watchStatus, 10 * 1000);
                 return;
               }
-              if (response.isInstalled) {
-                setEnableStatus({
-                  status: EnableApplicationStatus.SUCCESS,
-                  error: '',
-                });
-                dispatchResults(undefined);
-              }
+              setEnableStatus({
+                status:
+                  response.variablesValidationStatus === 'True'
+                    ? EnableApplicationStatus.SUCCESS
+                    : EnableApplicationStatus.FAILED,
+                error:
+                  response.variablesValidationStatus === 'True' ? '' : 'Variables are not valid',
+              });
+              dispatchResults(
+                response.variablesValidationStatus === 'True'
+                  ? undefined
+                  : 'Variables are not valid',
+              );
```

Comment:
Also the first time I am enabled NIM with the modal, it will tell me it failed. I believe this is because the `response.variablesValidationStatus` is `'Unknown'`

Reply:
@emilys314 The logic here is to check if `response.variablesValidationStatus` is true. If yes, `dispatchResults`. If the `response.variablesValidationStatus` is `Unknown`, it should loop again to check the status of the application.